backend/attendance-service/src/main/java/org/egov/repository/AttendanceRepository.java

```diff
@@ -20,48 +20,50 @@
 public class AttendanceRepository {
 
     @Autowired
-    private AttendeeQueryBuilder attendeeQueryBuilder;
+    AttendeeRepository attendeeRepository;
 
     @Autowired
-    private StaffQueryBuilder staffQueryBuilder;
-
-    @Autowired
-    private AttendeeRowMapper attendeeRowMapper;
-
-    @Autowired
-    private StaffRowMapper staffRowMapper;
+    StaffRepository staffRepository;
 
     @Autowired
     private JdbcTemplate jdbcTemplate;
 
     public List<AttendanceRegister> getAttendanceRegister(AttendanceRegisterSearchCriteria searchCriteria) {
-        List<Object> preparedStmtList = new ArrayList<>();
-        List<String> registerIds = new ArrayList<>();
 
-        String attendeeQuery = attendeeQueryBuilder.getAttendeeSearchQuery(searchCriteria, preparedStmtList);
-        List<AttendanceRegister> attendanceRegisters = jdbcTemplate.query(attendeeQuery, attendeeRowMapper, preparedStmtList.toArray());
+        List<String> registerIds;
+        List<AttendanceRegister> registerAttendees = new ArrayList<>();
+
+        if (StringUtils.isBlank(searchCriteria.getStaffId())) {
+            registerAttendees = attendeeRepository.getRegisteredAttendees(searchCriteria);
```

Comment:
This method's name doesn't clearly reflect what it is doing. It is returning AttendanceRegister and not Attendees. 
And, the AttendeeRepository should return you the Attendee-Register object and not the AttendanceRegister. 
The Attendee-Register object structure would be close to the table structure. So it would primarily have registerId and attendeeId, and the dates. 

The AttendeeReqpository will return you that. Now the AttendanceRegisterRepository will have to build the register from the those RegisterIds. 

With this approach, there would be 2 DB calls which is not the most efficient approach but if it is bringing more clarity and cleanliness to the code then I think it is worth the trade-off.

Reply:
The method returns the attendees registered to the attendance register. Hence the name getRegisteredAttendees. I feel 2 DB calls will result in lot of operations in the AttendanceRegisterRepository and also forbids the filtering at query level as search criteria includes registerNumber, name, fromDate , toDate which corresponds to the register.

Reply:
As discussed on call, simplify the repositories and make the necessary changes.